plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -985,6 +986,7 @@ public void rollback()
         TupleDomain<IcebergColumnHandle> newUnenforcedConstraint = constraint.getSummary()
                 .transformKeys(IcebergColumnHandle.class::cast)
                 .filter(isIdentityPartition.negate())
+                .filter((columnHandle, predicate) -> !isStructuralType(columnHandle.getType())) // TODO: Remove after completing https://github.com/trinodb/trino/issues/8759
```

Comment:
We should be able to calculate `newUnenforcedConstraint` from `constraint` and `newEnforcedConstraint` like in https://github.com/trinodb/trino/blob/b77dd122029a42bbf80ba9c9fd8853daf4522da9/core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java#L135-L143

then, as @phd3 suggested, it would be natural to apply the fix to the `newEnforcedConstraint` definition.

Reply:
We could, but since the enforced constraint either accepts an entire column domain or none of it this seems like more reasoning than it's worth compared to just negating the filter predicate.